From a2aadad0910ff4803d917dc29508037019e21222 Mon Sep 17 00:00:00 2001
From: Nobuhiko Tanibata <NOBUHIKO_TANIBATA@xddp.denso.co.jp>
Date: Tue, 15 Jul 2014 19:00:59 +0900
Subject: [PATCH 21/31] To get resolution of a screen, refer width and height
 of screen's of weston_output.

Priviously, this code refers width and height of current_mode. However it is not always available. E.g. on Tizen.

Signed-off-by: Nobuhiko Tanibata <NOBUHIKO_TANIBATA@xddp.denso.co.jp>
---
 ivi-shell/ivi-layout.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/ivi-shell/ivi-layout.c b/ivi-shell/ivi-layout.c
index 39b7a96..5ed5513 100644
--- a/ivi-shell/ivi-layout.c
+++ b/ivi-shell/ivi-layout.c
@@ -1576,8 +1576,10 @@ ivi_layout_getScreenResolution(struct ivi_layout_screen *iviscrn,
     }
 
     output   = iviscrn->output;
-    *pWidth  = output->current_mode->width;
-    *pHeight = output->current_mode->height;
+    weston_log("ivi-shell uses %s as a screen.\n", output->name);
+    *pWidth  = output->width;
+    *pHeight = output->height;
+    weston_log("ivi-shell: screen resolution is (%i,%i).\n", *pWidth,*pHeight);
 
     return 0;
 }
-- 
1.9.3

